### PR TITLE
lib/posix-user: Add README.md file for posix-user

### PR DESCRIPTION
**Pull request by DeliaPavel**
### Prerequisite checklist

 - [ ] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
 - [ ] Tested your changes against relevant architectures and platforms;
 - [ ] Ran the [`checkpatch.uk`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.uk) on your commit series before opening this PR;
 - [ ] Updated relevant documentation.

### Base target

 - Architecture(s): [e.g. `x86_64` or N/A]
 - Platform(s): [e.g. `kvm`, `xen` or N/A]
 - Application(s): [e.g. `app-python3` or N/A]

### CODE REVIEW

File: lib/posix-user/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ .pw_passwd = UK_DEFAULT_PASS,
+ .pw_uid = UK_DEFAULT_UID,
+ .pw_gid = UK_DEFAULT_GID,
+ .pw_gecos = UK_DEFAULT_USER,
+ .pw_dir = "/",
+ .pw_shell = "",
+};
+```
+
+The `passwd` structure is used to define the default `passwd` entry, whose format is similar to the ones from the `etc/passwd` file in Linux:
```
Comment (StefanJum) [inline lib/posix-user/README.md]:
```suggestion
The `passwd` structure is used to define the default `passwd` entry, whose format is similar to the one from the `/etc/passwd` file in Linux:
```

File: lib/posix-user/README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+* `pw_name` refers to the name of the user, `UK_DEFAULT_USER`, which is actually `CONFIG_LIBPOSIX_USER_USERNAME`.
+* `pw_passwd` is the password of the default user, which is the empty string (`""`), meaning that the user does not have a password.
+* `pw_uid` is the user ID of the default user, which is equal to `CONFIG_LIBPOSIX_USER_UID`.
+* `pw_gid` is the ID of the default user's group and is equal to
+`CONFIG_LIBPOSIX_USER_GID`.
```
Comment (StefanJum) [inline lib/posix-user/README.md]:
```suggestion
* `pw_gid` is the ID of the default user's group and is equal to `CONFIG_LIBPOSIX_USER_GID`.
```

File: lib/posix-user/README.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+ .gr_mem = g_members__,
+};
+```
+
+The `group` structure is used to define the default `group` entry, whose format is similar to the ones from the `etc/group` file in Linux:
+
+* `gr_name` refers to the name of the group, `UK_DEFAULT_GROUP`, which is actually `CONFIG_LIBPOSIX_USER_GROUPNAME`.
+* `gr_passwd` is the password of the default group, which is the empty string (`""`), meaning that the group does not have a password.
+* `gr_gid` is the ID of the default group and is equal to
+`CONFIG_LIBPOSIX_USER_GID`.
```
Comment (StefanJum) [inline lib/posix-user/README.md]:
```suggestion
* `gr_gid` is the ID of the default group and is equal to `CONFIG_LIBPOSIX_USER_GID`.
```

**Review by StefanJum — CHANGES_REQUESTED** [review]
Looks good @DeliaPavel, thanks.
Besides the comments, see if you can please the codacy linter.

Also please add instructions on how to set the config variables (CONFIG_LIBPOSIX_USER...) and specify their default values.

**Review by DeliaPavel — COMMENTED** [review]
I tried pleasing the Codacy linter, but I don't really understand the comments we have left. Also, for the `[ordered-list-marker-value]`, I know for past PRs we used 1. for all the list values.

**Review by razvand — APPROVED** [review]
Reviewed-by: Razvan Deaconescu <razvand@unikraft.io>
Approved-by: Razvan Deaconescu <razvand@unikraft.io>

**Pull request merged**